tools/dart_tools/lib/runCommand.dart

```diff
@@ -0,0 +1,18 @@
+import 'dart:io';
+
+Future<int> runCommand(String exe, List<String> arguments,
+    {String workingDirectory, bool allowFail = false}) async {
+  final fullCommand = '$exe ${arguments.join(" ")}';
+  print('Running $fullCommand');
+
+  final process = await Process.start(exe, arguments,
+      workingDirectory: workingDirectory, runInShell: true);
+  stdout.addStream(process.stdout);
+  stderr.addStream(process.stderr);
+
+  final exitCode = await process.exitCode;
+  if (!allowFail && exitCode != 0) {
+    throw new Exception('$fullCommand failed with exit code $exitCode');
+  }
+  return exitCode;
+}
```

Comment:
Nit: please add missing newline. (So editors will auto-add them if missing, so it'll cause random churn later if checked in this way.)